backend/src/main/kotlin/com/ci5644/trade/controllers/api/UserController.kt

```diff
@@ -0,0 +1,72 @@
+package com.ci5644.trade.controllers.api
+
+import com.ci5644.trade.services.user.UserService
+import com.ci5644.trade.dto.UserDto
+import com.ci5644.trade.dto.UserDetailsDto
+import com.ci5644.trade.config.SecurityConstants
+import com.ci5644.trade.services.auth.AuthorizationService
+import com.ci5644.trade.config.JWT.JWTSecurityUtils
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+
+/**
+ * Controller class to handle user-related HTTP requests.
+ * Uses JWT authentication for requests and CORS configuration.
+ */
+@RestController
+@RequestMapping("/api/user")
```

Comment:
If every endpoint will start with api, have it configured in the base path

Also this should be users